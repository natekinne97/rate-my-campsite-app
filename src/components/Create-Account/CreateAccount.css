.account-form button{
	width: 25%;
	margin-top: 2%;
}

.new-site button{
	width: 25%;
	margin-top: 2%;
}
