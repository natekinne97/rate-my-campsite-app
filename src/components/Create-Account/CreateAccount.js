import React from 'react';

class CreateAccount extends React.Component {

    render() {
        return (
            <div>
                <header>Create Account</header>
                <form className="account-form gen-form">

                    <label htmlFor="username">Username</label>
                    <input id="username" type="text" name="username" required/>

                    <label htmlFor="profile-pic">Profile picture</label>
                    <input type="file" name="profile-pic"/>

                    <label htmlFor="password">Password</label>
                    <input id="password" type="password" name="password" required/>

                    <label htmlFor="repeat-password">Repeat password</label>
                    <input id="repeat-password" type="password" name="repeat-password" required/>

                    <button type="submit">Sign Up</button>

	        	</form>
            </div>
        );
    }
}

export default CreateAccount;