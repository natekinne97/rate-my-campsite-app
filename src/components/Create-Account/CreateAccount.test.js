import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CreateAccount from './CreateAccount';

describe.only('Review form component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<CreateAccount />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<CreateAccount />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the UI as expected with no unreads', () => {
        const tree = renderer
            .create(<CreateAccount />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});