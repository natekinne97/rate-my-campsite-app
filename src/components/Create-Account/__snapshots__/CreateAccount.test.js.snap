// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review form component renders the UI as expected 1`] = `
<div
  className="create-account"
>
  <header>
    Create Account
  </header>
  <form
    className="account-form gen-form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="username"
    >
      Username
    </label>
    <input
      id="username"
      name="user_name"
      required={true}
      type="text"
    />
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      name="email"
      required={true}
      type="email"
    />
    <label
      htmlFor="first_name"
    >
      First name
    </label>
    <input
      name="first_name"
      required={true}
      type="text"
    />
    <label
      htmlFor="last_name"
    >
      Last name
    </label>
    <input
      name="last_name"
      type="text"
    />
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      required={true}
      type="password"
    />
    <label
      htmlFor="repeat-password"
    >
      Repeat password
    </label>
    <input
      id="repeat-password"
      name="repeat_password"
      onChange={[Function]}
      required={true}
      type="password"
    />
    <button
      type="submit"
    >
      Sign Up
    </button>
  </form>
</div>
`;

exports[`Review form component renders the UI as expected with no unreads 1`] = `
<div
  className="create-account"
>
  <header>
    Create Account
  </header>
  <form
    className="account-form gen-form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="username"
    >
      Username
    </label>
    <input
      id="username"
      name="user_name"
      required={true}
      type="text"
    />
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      name="email"
      required={true}
      type="email"
    />
    <label
      htmlFor="first_name"
    >
      First name
    </label>
    <input
      name="first_name"
      required={true}
      type="text"
    />
    <label
      htmlFor="last_name"
    >
      Last name
    </label>
    <input
      name="last_name"
      type="text"
    />
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      required={true}
      type="password"
    />
    <label
      htmlFor="repeat-password"
    >
      Repeat password
    </label>
    <input
      id="repeat-password"
      name="repeat_password"
      onChange={[Function]}
      required={true}
      type="password"
    />
    <button
      type="submit"
    >
      Sign Up
    </button>
  </form>
</div>
`;
